dist/PreserveModuleNamePlugin.js

```diff
@@ -73,6 +73,29 @@ function getPreservedModules(modules) {
                 m[exports.preserveModuleName] = req;
                 return true;
             }
+            // Since its relative, link it with the module alias if possible
```

Comment:
`getPreservedModules` is not really the good function for that. It's goal is to return the list of modules that must have their ids preserved.

Finding good ids for those modules starts later, on line 39.
There are several variations, you can add a new one, or maybe see if yours fit into `aliasRelative`, which already exists (but takes lower precedence than a path in a `modules` folder).